libskale/OverlayDB.cpp

```diff
@@ -95,49 +97,74 @@ dev::h256 OverlayDB::getLastExecutedTransactionHash() const {
+            const std::string l( value.begin(), value.end() );
+            if ( !l.empty() ) {
+                dev::bytes b( l.begin(), l.end() );
+                partialTransactionReceipts.push_back( b );
+            }
+            return true;
+        } );
     }
-
-    lastExecutedTransactionReceipts = partialTransactionReceipts;
     return partialTransactionReceipts;
 }
 
+
+void OverlayDB::removeAllPartialTransactionReceipts() {
+    // first we get all keys
+
+    string prefix( "safeLastTransactionReceipts." );
+    vector< string > keys;
+    if ( m_db_face ) {
+        m_db_face->forEachWithPrefix( prefix, [&keys]( Slice key, Slice ) {
+            const std::string keyStr( key.begin(), key.end() );
+            keys.push_back( keyStr );
+            return true;
+        } );
+
+        for ( auto&& key : keys ) {
+            // now remove all of them
+            m_db_face->kill( key );
+        }
+    }
+
+    m_db_face->commit( "Clean partial keys" );
+}
+
+
 void OverlayDB::setLastExecutedTransactionHash( const dev::h256& _newHash ) {
     this->lastExecutedTransactionHash = _newHash;
 }
-void OverlayDB::setPartialTransactionReceipts( const dev::bytes& _newReceipts ) {
-    this->lastExecutedTransactionReceipts = _newReceipts;
-}
 
-void OverlayDB::addReceiptToPartials( const dev::eth::TransactionReceipt& _receipt ) {
-    auto rawTransactionReceipts = getPartialTransactionReceipts();
+// transform uin564_t to hex format of fixed length prepending zeros
+// if needed. This is needed to put keys into LevelDB in lexicographic order
+std::string OverlayDB::uint64ToFixedLengthHex( uint64_t value ) {
+    // Create a stringstream to hold the hex representation
+    std::stringstream res;
 
-    // TODO Temporary solution - do not (de)serialize forth and back!
+    // Set formatting options: hex, fill with zeros, and a width of 16 characters
+    // (uint64_t can have a maximum of 16 hex characters).
+    res << std::hex << std::setw( 16 ) << std::setfill( '0' ) << value;
 
-    dev::eth::BlockReceipts blockReceipts;
-    if ( !rawTransactionReceipts.empty() ) {
-        dev::RLP rlp( rawTransactionReceipts );
-        blockReceipts = dev::eth::BlockReceipts( rlp );
-    }  // if
+    // Convert the stream to a string
+    return res.str();
+}
 
-    blockReceipts.receipts.push_back( _receipt );
+void OverlayDB::setPartialTransactionReceipt( const dev::bytes& _newReceipt,
+    dev::eth::BlockNumber _blockNumber, uint64_t _transactionIndex ) {
+    string key = "safeLastTransactionReceipts." + uint64ToFixedLengthHex( _blockNumber ) + "." +
+                 uint64ToFixedLengthHex( _transactionIndex );
 
-    setPartialTransactionReceipts( blockReceipts.rlp() );
-}
 
-void OverlayDB::clearPartialTransactionReceipts() {
-    dev::eth::BlockReceipts blockReceipts;
-    setPartialTransactionReceipts( blockReceipts.rlp() );
+    m_db_face->insert( skale::slicing::toSlice( key ), skale::slicing::toSlice( _newReceipt ) );
```

Comment:
why cant we do `this->lastExecutedTransactionReceipt = _newRecipt;` here and then commit this data during the regular `commit` stage? in this way we will do twice less inserts to the db per block